src/installer/managed/Microsoft.NET.HostModel/Bundle/Manifest.cs

```diff
@@ -65,7 +65,7 @@ private enum HeaderFlags : ulong
         // identify this bundle. It is choosen to be compatible
         // with path-names so that the AppHost can use it in
         // extraction path.
-        public readonly string BundleID;
+        public string BundleID { get; internal set; } = Path.GetRandomFileName();
```

Comment:
Is this `GetRandomFileName` used? Would it break determinism?

Reply:
Yes and yes :-) But not in runtime usages.
The `GetRandomFileName` was used originally and was the source of the issue this PR is trying to address. At the same time `BundleID` is a public member in a public class that is part of publicly available nuget (`Microsoft.NET.HostModel.nupkg`) - so I was considering two options how to address the issue while not breaking the compatibility:
1) Initialize the `BundleID` on first access and then throw if internal state of `Manifest` would be changing (call to `AddEntry`)
2) Keep the original functionality but give the internal callers (`Bundler`) the ability to specify the id.

Then There is a fact that `Manifest` don't have all necessary info the properly generate deterministic id with good distribution (as it has just info about file relative paths and lengths), so I'd need to adjust the contract for internal callers anyways.

Given those facts I decided to keep the same behavior for any possible external caller without throwing (as this functionality is likely not intended for external use anyway and published as nuget just by mistake: https://github.com/dotnet/runtime/issues/32825)  and fix the behavior for internal callers with less invasive change.

Please let me know if you'd preffer different approach

Reply:
I wouldn't worry about breaking changes in this case. As you mention, the NuGet publishing is historical and should go away - if there's a dependency on it somewhere, I would actually prefer to find it (and break it :-)).

For the same reasons, if it doesn't make sense to have the property here or if it needs to be changed, just do it.

Reply:
Sweet:-)
OK - I'll clean this change a bit. Self identification should be responsibility of `Manifest`